src/lib/tracking.ts

```diff
@@ -0,0 +1,20 @@
+import { EpicMetadata } from '../components/ContributionsEpic';
+
+export const getTrackingUrl = (baseUrl: string, params: EpicMetadata): string => {
+    const acquisitionData = encodeURIComponent(
+        JSON.stringify({
+            source: params.platformId,
+            componentId: params.campaignCode,
+            componentType: params.ophanComponentId,
+            campaignCode: params.campaignCode,
+            abTest: {
+                name: params.abTestName,
+                variant: params.abTestVariant,
+            },
+            referrerPageviewId: params.ophanPageId,
+            referrerUrl: params.referrerUrl,
+        }),
+    );
+
+    return `${baseUrl}?REFPVID=${params.ophanPageId}&INTCMP=${params.campaignCode}&acquisitionData=${acquisitionData}`;
```

Comment:
Is it worth for the query params building a map and then joining the parts together? It's a small thing, but might help with readability and easier changes in the future.

Reply:
Good shout, I've refactored this to map through the entries of an object containing all parts of the query string and turning each into a key=value string, then joining all of these to end up with the complete query string.